src/TriggerBinding/SqlTriggerMetricsProvider.cs

```diff
@@ -0,0 +1,130 @@
+            this._userFunctionId = !string.IsNullOrEmpty(userFunctionId) ? userFunctionId : throw new ArgumentNullException(nameof(userFunctionId));
+        }
+        public async Task<SqlTriggerMetrics> GetMetricsAsync()
+        {
+            return new SqlTriggerMetrics
+            {
+                UnprocessedChangeCount = await this.GetUnprocessedChangeCountAsync(),
+                Timestamp = DateTime.UtcNow,
+            };
+        }
+        private async Task<long> GetUnprocessedChangeCountAsync()
+        {
+            long unprocessedChangeCount = 0L;
+
+            try
+            {
+                long getUnprocessedChangesDurationMs = 0L;
+
+                using (var connection = new SqlConnection(this._connectionString))
+                {
+                    this._logger.LogDebugWithThreadId("BEGIN OpenGetUnprocessedChangesConnection");
+                    await connection.OpenAsync();
+                    this._logger.LogDebugWithThreadId("END OpenGetUnprocessedChangesConnection");
+
+                    int userTableId = await GetUserTableIdAsync(connection, this._userTable.QuotedFullName, this._logger, CancellationToken.None);
+                    IReadOnlyList<(string name, string type)> primaryKeyColumns = await GetPrimaryKeyColumnsAsync(connection, userTableId, this._logger, this._userTable.FullName, CancellationToken.None);
+
+                    // Use a transaction to automatically release the app lock when we're done executing the query
+                    using (SqlTransaction transaction = connection.BeginTransaction(IsolationLevel.RepeatableRead))
+                    {
+                        try
+                        {
+                            using (SqlCommand getUnprocessedChangesCommand = this.BuildGetUnprocessedChangesCommand(connection, transaction, primaryKeyColumns, userTableId))
+                            {
+                                this._logger.LogDebugWithThreadId($"BEGIN GetUnprocessedChangeCount Query={getUnprocessedChangesCommand.CommandText}");
+                                var commandSw = Stopwatch.StartNew();
+                                unprocessedChangeCount = (long)await getUnprocessedChangesCommand.ExecuteScalarAsync();
+                                getUnprocessedChangesDurationMs = commandSw.ElapsedMilliseconds;
+                            }
+
+                            this._logger.LogDebugWithThreadId($"END GetUnprocessedChangeCount Duration={getUnprocessedChangesDurationMs}ms Count={unprocessedChangeCount}");
+                            transaction.Commit();
+                        }
+                        catch (Exception ex)
+                        {
+                            TelemetryInstance.TrackException(TelemetryErrorName.GetUnprocessedChangeCount, ex, null, new Dictionary<TelemetryMeasureName, double>() { { TelemetryMeasureName.GetUnprocessedChangesDurationMs, getUnprocessedChangesDurationMs } });
```

Comment:
We can probably remove this one since you're sending it below too. Make sure you add the duration to that one though.